pkg/controller/perconaservermongodb/users.go

```diff
@@ -60,52 +59,105 @@ func (r *ReconcilePerconaServerMongoDB) reconcileUsers(cr *api.PerconaServerMong
 		}
 		err = r.client.Create(context.TODO(), internalSysUsersSecret)
 		if err != nil {
-			return nil, nil, errors.Wrap(err, "create internal sys users secret")
+			return errors.Wrap(err, "create internal sys users secret")
 		}
-		return nil, nil, nil
+		return nil
 	}
 
 	// we do this check after work with secret objects because in case of upgrade cluster we need to be sure that internal secret exist
 	if cr.Status.State != api.AppStateReady {
-		return nil, nil, nil
+		return nil
 	}
 
 	newSysData, err := json.Marshal(sysUsersSecretObj.Data)
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "marshal sys secret data")
+		return errors.Wrap(err, "marshal sys secret data")
 	}
+
 	newSecretDataHash := sha256Hash(newSysData)
 	dataChanged, err := sysUsersSecretDataChanged(newSecretDataHash, &internalSysSecretObj)
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "check sys users data changes")
+		return errors.Wrap(err, "check sys users data changes")
 	}
 
 	if !dataChanged {
-		return nil, nil, nil
+		return nil
 	}
 
 	restartSfs, restartMongos, err := r.updateSysUsers(cr, &sysUsersSecretObj, &internalSysSecretObj, repls)
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "manage sys users")
+		return errors.Wrap(err, "manage sys users")
+	}
+
+	containers := []string{"pmm-client", "backup-agent"}
```

Comment:
why we need to restart pmm if only the backup password was changed and vise versa